cirq/google/calibration/workflow.py

```diff
@@ -337,19 +351,150 @@ def run_characterizations(
     return results
 
 
-@dataclasses.dataclass(frozen=True)
-class CircuitPhasedFSimCalibrationResults:
-    """Circuit-specific calibration results
+def zeta_chi_gamma_calibration_for_moments(
+    circuit_calibration: CircuitCalibration,
+    characterizations: List[PhasedFSimCalibrationResult],
+    gates_translator: Callable[[Gate], Optional[FSimGate]] = try_convert_sqrt_iswap_to_fsim,
+) -> CircuitCalibration:
+    """Compensates circuit against errors in zeta, chi and gamma angles.
+
+    This method creates a new circuit with a single-qubit Z gates added in a such way so that
+    zeta, chi and gamma angles discovered by characterizations are cancelled-out and set to 0.
+
+    This function preserves a moment structure of the circuit. All single qubit gates appear on new
+    moments in the final circuit.
+
+    Args:
+        circuit_calibration: Description of the circuit together with its calibration metadata that
```

Comment:
"A `CircuitRequestMapper` (likely returned from `run_calibrations`) whose mapping argument corresponds to the results in the `characterizations` argument"

^^ suggested wording, but you should flesh out, make sure it's accurate, and replace the names with whatever you settle on. I find it helpful to spell out explicitly exactly how these parts fit together.

Reply:
Done.